src/bosh-softlayer-cpi/softlayer/common/vm_utils.go

```diff
@@ -143,21 +142,29 @@ func ParseMbusURL(mbusURL string, primaryBackendIpAddress string) (string, error
 	return fmt.Sprintf("%s://%s:%s", parsedURL.Scheme, primaryBackendIpAddress, port), nil
 }
 
-func UpdateEtcHostsOfBoshInit(fileName string, record string) (err error) {
-	buffer := bytes.NewBuffer([]byte{})
-	t := template.Must(template.New("etc-hosts").Parse(ETC_HOSTS_TEMPLATE))
+func UpdateEtcHostsOfBoshInit(path string, record string) (err error) {
+	logger := boshlog.NewWriterLogger(boshlog.LevelError, os.Stderr, os.Stderr)
+	fs := boshsys.NewOsFileSystem(logger)
 
-	err = t.Execute(buffer, record)
+	if !fs.FileExists(path) {
+		err := fs.WriteFile(path, []byte{})
+		if err != nil {
+			return bosherr.WrapErrorf(err, "Creating the new file %s if it does not exist", path)
+		}
+	}
+
+	fileHandle, err := fs.OpenFile(path, os.O_APPEND|os.O_WRONLY, os.ModeAppend)
 	if err != nil {
-		return bosherr.WrapError(err, "Generating config from template")
+		return bosherr.WrapErrorf(err, "Opening file %s", path)
 	}
 
-	logger := boshlog.NewWriterLogger(boshlog.LevelError, os.Stderr, os.Stderr)
-	fs := boshsys.NewOsFileSystem(logger)
+	writer := bufio.NewWriter(fileHandle)
+	defer fileHandle.Close()
 
-	err = fs.WriteFile(fileName, buffer.Bytes())
+	fmt.Fprintln(writer, "\n"+record)
```

Comment:
Why are you not checking for errors writing the `writer`? 

The following call: `fmt.Fprintln(writer, "\n"+record)` should return an error and the number of bytes written. There should be no errors and the number of bytes written should be equal to the number of bytes in record plus new line. Please ensure that the necessary checks are done and appropriate tests are added for these possible failure paths.

Thanks.